@/components/ui/calendar.tsx

```diff
@@ -1,70 +1,153 @@
+      Dropdown: (props) => {
+        const { fromDate, fromMonth, fromYear, toDate, toMonth, toYear } =
+          useDayPicker();
+        const { goToMonth, currentMonth } = useNavigation();
+
+        if (props.name === 'months') {
+          const selectedItems = Array.from({ length: 12 }, (_, i) => ({
+            value: i.toString(),
+            label: format(setMonth(new Date(), i), 'MMMM', { locale }),
+          }));
+
+          return (
+            <Select
+              onValueChange={(newValue) => {
+                const newDate = new Date(currentMonth);
+                newDate.setMonth(parseInt(newValue, 10));
+                goToMonth(newDate);
+              }}
+            >
+              <SelectTrigger className="p-0 m-0 border-0 shadow-none outline-0">
+                {format(currentMonth, 'MMMM', { locale })}
+              </SelectTrigger>
+              <SelectContent>
+                {selectedItems.map((selectItem) => (
+                  <SelectItem value={selectItem.value}>
+                    {selectItem.label}
+                  </SelectItem>
+                ))}
+              </SelectContent>
+            </Select>
+          );
+        }
+        if (props.name === 'years') {
+          const earliestYear =
+            fromYear || fromMonth?.getFullYear() || fromDate?.getFullYear();
+          const latestYear =
+            toYear || toMonth?.getFullYear() || toDate?.getFullYear();
+
+          if (earliestYear && latestYear) {
+            const yearsLength = latestYear - earliestYear + 1;
+
+            const selectedItems = Array.from(
+              { length: yearsLength },
+              (_, i) => ({
+                value: (earliestYear + i).toString(),
+                label: (earliestYear + i).toString(),
+              })
+            );
+
+            return (
+              <Select
+                onValueChange={(newValue) => {
+                  const newDate = new Date(currentMonth);
+                  newDate.setFullYear(parseInt(newValue, 10));
+                  goToMonth(newDate);
+                }}
+                value={props.value?.toString()}
+              >
+                <SelectTrigger className="p-0 m-0 border-0 shadow-none outline-0">
+                  {currentMonth.getFullYear()}
+                </SelectTrigger>
+                <SelectContent>
+                  {selectedItems.map((selectItem) => (
+                    <SelectItem value={selectItem.value}>
+                      {selectItem.label}
+                    </SelectItem>
+                  ))}
+                </SelectContent>
+              </Select>
+            );
+          }
+        }
+        return null;
+      },
+      // eslint-disable-next-line react/no-unstable-nested-components
+      IconLeft: () => <ChevronLeftIcon className="h-4 w-4" />,
+      // eslint-disable-next-line react/no-unstable-nested-components
+      IconRight: () => <ChevronRightIcon className="h-4 w-4" />,
```

Comment:
We tend to follow the eslint standards, can you solve the issues provided in it? Here is a link about the issue: 
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unstable-nested-components.md